From: Simon McVittie <smcv@debian.org>
Date: Tue, 21 Sep 2021 12:20:20 +0100
Subject: tests: Don't use TestEnvironment

The specification for installed-tests says it's an overlay onto the
current environment, but the implementation in gnome-desktop-testing
currently removes all other environment variables, causing test failure
(see GNOME/gnome-desktop-testing#1).

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 cogl/tests/conform/mutter-cogl.test.in | 3 +--
 src/tests/mutter-all.test.in           | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/cogl/tests/conform/mutter-cogl.test.in b/cogl/tests/conform/mutter-cogl.test.in
index fae4f5c..7a8cd13 100644
--- a/cogl/tests/conform/mutter-cogl.test.in
+++ b/cogl/tests/conform/mutter-cogl.test.in
@@ -1,4 +1,3 @@
 [Test]
 Type=session
-TestEnvironment=COGL_TEST_VERBOSE=1
-Exec=sh -c "cd @libexecdir@/installed-tests/mutter-@apiversion@/cogl/conform; xvfb-run -a -s '+iglx -noreset' ./run-tests.sh ./config.env ./test-conformance ./unit-tests"
+Exec=sh -c "export COGL_TEST_VERBOSE=1; cd @libexecdir@/installed-tests/mutter-@apiversion@/cogl/conform; xvfb-run -a -s '+iglx -noreset' ./run-tests.sh ./config.env ./test-conformance ./unit-tests"
diff --git a/src/tests/mutter-all.test.in b/src/tests/mutter-all.test.in
index 413d2ac..c6b06a9 100644
--- a/src/tests/mutter-all.test.in
+++ b/src/tests/mutter-all.test.in
@@ -1,6 +1,5 @@
 [Test]
 Description=All Mutter tests
-TestEnvironment=GSETTINGS_BACKEND=memory;
-Exec=sh -c 'env XDG_RUNTIME_DIR="$(mktemp -d -t mutter-@apiversion@-all-tests-XXXXXX)" @libexecdir@/installed-tests/mutter-@apiversion@/meta-dbus-runner.py xvfb-run -a -s "+iglx -noreset" -- @libexecdir@/installed-tests/mutter-@apiversion@/mutter-test-runner --all'
+Exec=sh -c 'env GSETTINGS_BACKEND=memory XDG_RUNTIME_DIR="$(mktemp -d -t mutter-@apiversion@-all-tests-XXXXXX)" @libexecdir@/installed-tests/mutter-@apiversion@/meta-dbus-runner.py xvfb-run -a -s "+iglx -noreset" -- @libexecdir@/installed-tests/mutter-@apiversion@/mutter-test-runner --all'
 Type=session
 Output=TAP
